onnxruntime/core/providers/mkldnn/math/sum.cc

```diff
@@ -0,0 +1,232 @@
+// Copyright(C) 2018 Intel Corporation
+// Licensed under the MIT License
+
+#ifdef _WIN32
+#pragma warning(disable : 4244)
+#endif
+
+#include "core/providers/mkldnn/mkldnn_common.h"
+#include "core/providers/mkldnn/math/sum.h"
+#include "core/providers/mkldnn/mkldnn_fwd.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+namespace {
+// Struct which encapsulates parameters for MKLDNN Sum primitives.
+struct SumParams {
+  std::vector<mkldnn::memory::dims> src_dims;
+  mkldnn::memory::dims& dst_dim;
+  int num_inputs;
+  int num_dimensions;
+
+  SumParams(std::vector<mkldnn::memory::dims> dims,
+    mkldnn::memory::dims& dst_dims, int &numinputs, int &dimensions)
```

Comment:
any reason you're passing numinputs and dimensions by ref?

Reply:
yes. numinputs and dimensions should be pass by value. I will update it.